### ISSUE TITLE
Qt 5.9

### ISSUE THREAD

**Issue by joeblew99** [therecipe/qt#392]
Have been using 5.9 for a month now against this code and no issues btw.
I use it because I wanted watch support as well as desktop and mobile, and 5.9 had some things I needed.

I read on Qt web site that it's long term support too.

Anyway just wanted to tell everyone if this helps.

**Comment by RadhiFadlillah** [comment]
@joeblew99 @therecipe unfortunately 5.9.0 is still not working for me. When I try to build my app, it failed with following message :

```
ERRO[0021] failed to run command cmd="go build -p 4 -v -ldflags="-s" "-w" -o $HOME/GoProjects/src/my-project/deploy/linux/my-project -i -tags=""" env= error="exit status 2" func=RunCmd name="build for linux on linux"
my-project
# my-project
/usr/lib/go/pkg/tool/linux_amd64/link: running g++ failed: exit status 1
/tmp/go-link-453244066/000003.o: In function `QFileDialog_SelectedMimeTypeFilter':
$HOME/GoProjects/src/github.com/therecipe/qt/widgets/widgets.cpp:12805: undefined reference to `QFileDialog::selectedMimeTypeFilter() const'
/tmp/go-link-453244066/000003.o: In function `QSplitter_ReplaceWidget':
$HOME/GoProjects/src/github.com/therecipe/qt/widgets/widgets.cpp:41522: undefined reference to `QSplitter::replaceWidget(int, QWidget*)'
/tmp/go-link-453244066/000003.o: In function `QSystemTrayIcon_ShowMessage2Default':
$HOME/GoProjects/src/github.com/therecipe/qt/widgets/widgets.cpp:47501: undefined reference to `QSystemTrayIcon::showMessage(QString const&, QString const&, QIcon const&, int)'
/tmp/go-link-453244066/000003.o: In function `QUndoCommand_SetObsolete':
$HOME/GoProjects/src/github.com/therecipe/qt/widgets/widgets.cpp:54908: undefined reference to `QUndoCommand::setObsolete(bool)'
/tmp/go-link-453244066/000003.o: In function `QUndoCommand_IsObsolete':
$HOME/GoProjects/src/github.com/therecipe/qt/widgets/widgets.cpp:54949: undefined reference to `QUndoCommand::isObsolete() const'
/tmp/go-link-453244066/000003.o: In function `QWidget_SetWindowFlag':
$HOME/GoProjects/src/github.com/therecipe/qt/widgets/widgets.cpp:63294: undefined reference to `QWidget::setWindowFlag(Qt::WindowType, bool)'
/tmp/go-link-453244066/000006.o: In function `QByteArray_ToHex2':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:4180: undefined reference to `QByteArray::toHex(char) const'
/tmp/go-link-453244066/000006.o: In function `QDir_IsEmpty':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:6614: undefined reference to `QDir::isEmpty(QFlags<QDir::Filter>) const'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_Name':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14391: undefined reference to `QOperatingSystemVersion::name() const'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_Current':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14416: undefined reference to `QOperatingSystemVersion::current()'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_AndroidJellyBean':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14421: undefined reference to `QOperatingSystemVersion::AndroidJellyBean'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_AndroidJellyBean_MR1':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14426: undefined reference to `QOperatingSystemVersion::AndroidJellyBean_MR1'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_AndroidJellyBean_MR2':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14431: undefined reference to `QOperatingSystemVersion::AndroidJellyBean_MR2'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_AndroidKitKat':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14436: undefined reference to `QOperatingSystemVersion::AndroidKitKat'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_AndroidLollipop':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14441: undefined reference to `QOperatingSystemVersion::AndroidLollipop'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_AndroidLollipop_MR1':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14446: undefined reference to `QOperatingSystemVersion::AndroidLollipop_MR1'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_AndroidMarshmallow':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14451: undefined reference to `QOperatingSystemVersion::AndroidMarshmallow'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_AndroidNougat':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14456: undefined reference to `QOperatingSystemVersion::AndroidNougat'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_AndroidNougat_MR1':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14461: undefined reference to `QOperatingSystemVersion::AndroidNougat_MR1'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_MacOSSierra':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14466: undefined reference to `QOperatingSystemVersion::MacOSSierra'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_OSXElCapitan':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14471: undefined reference to `QOperatingSystemVersion::OSXElCapitan'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_OSXMavericks':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14476: undefined reference to `QOperatingSystemVersion::OSXMavericks'
/tmp/go-link-453244066/000006.o: In function `QOperatingSystemVersion_QOperatingSystemVersion_OSXYosemite':
$HOME/GoProjects/src/github.com/therecipe/qt/core/core.cpp:14481: undefined reference to `QOperatingSystemVersion::OSXYosemite'
/tmp/go-link-453244066/000006.o: InThe terminal process terminated with exit code: 1
```

It seems these error happened because [new classes and functions](https://doc.qt.io/qt-5/newclasses59.html) in Qt 5.9 has not implemented yet in this package.

**Comment by jmc-figueira** [comment]
In my case I also get an error running `qtsetup`, but in my case it appears to be related to having Qt 5.9.1 installed, which Qt recognizes as 5.9 (the folder for which doesn't exist).

`Queimado-Mac:GoProjects joao$ echo $QT_VERSION_MAJOR
5.9.1
Queimado-Mac:GoProjects joao$ /Users/joao/Documents/GoProjects/bin/qtsetup
INFO[0000] running: 'qtsetup prep'
INFO[0000] successfully created qtrcc symlink in your PATH (/usr/local/bin/qtrcc)
INFO[0000] successfully created qtmoc symlink in your PATH (/usr/local/bin/qtmoc)
INFO[0000] successfully created qtminimal symlink in your PATH (/usr/local/bin/qtminimal)
INFO[0000] successfully created qtdeploy symlink in your PATH (/usr/local/bin/qtdeploy)
INFO[0000] running: 'qtsetup check darwin' [docker=false]
INFO[0000] GOOS:                        'darwin'
INFO[0000] GOARCH:                      'amd64'
INFO[0000] GOVERSION:                   'go1.8'
INFO[0000] GOROOT:                      '/usr/local/go'
INFO[0000] GOPATH:                   *  '/Users/joao/Documents/GoProjects'
INFO[0000] GOBIN:                       '/Users/joao/Documents/GoProjects/bin'
INFO[0000] QT_HASH:                     '7ca03f7c288a2207fa524765b1c101f9e7c97693'
INFO[0000] QT_VERSION:               *  '5.9.1'
INFO[0000] QT_VERSION_MAJOR:         *  '5.9'`

Followed by:

`ERRO[0000] failed to load /Users/joao/Documents/Qt/Docs/Qt-5.9/qtcore/qtcore.index  error="open /Users/joao/Documents/Qt/Docs/Qt-5.9/qtcor
e/qtcore.index: no such file or directory"`

**Comment by therecipe** [comment]
@RadhiFadlillah  @jmc-figueira 
Did you re-ran the setup with `QT_DIR=path/to/Qt5.9.0` and `QT_VERSION=5.9.0` (or `5.9.1`) ?
You will need to make these env variables persistent.

I just tested 5.9 on macOS, but haven't tested if it works on linux/windows yet.
However, I will work on that in the next few days.

**Comment by RadhiFadlillah** [comment]
@therecipe I've only tried setting `QT_DIR` and `QT_VERSION` from `qtsetup` flags like this :

```
qtsetup -qt_dir "$HOME/Qt5.9.0" -qt_version "5.9.0"
```

I will try to set these variables to persistent.

**Comment by RadhiFadlillah** [comment]
@therecipe after I set `QT_DIR` and `QT_VERSION` in `.profile`, Qt 5.9.0 works succesfully 👍👍👍

Like @jmc-figueira, I also tried using Qt 5.9.1, but it's failed even after I set those environment variables. It seems it's failed because in Qt 5.9.1 some directory has changed :

- `Qt5.9.1/Docs/Qt-5.9` is not exist, replaced by `Qt5.9.1/Docs/Qt-5.9.1`
- `Qt5.9.1/5.9` is not exist, replaced by `Qt5.9.1/5.9.1`

After I rename those two directory, `qtsetup` finished succesfully. My app is also build and run succesfully using `qtdeploy`.

**Comment by therecipe** [comment]
@RadhiFadlillah 
Fixed the issue with 5.9.1: https://github.com/therecipe/qt/commit/9127474fe466369177aeef9e78ca8c547411eedc
But, I haven't tested it on linux and windows yet.

**Comment by RadhiFadlillah** [comment]
Thanks @therecipe, I can confirm it works in Manjaro Linux 64-bit.

Btw in Qt 5.9+ there is [bug](https://bugreports.qt.io/browse/QTBUG-61395) where Alt+Hotkey is not working on QMessageBox and QInputDialog, so for now I will keep using Qt 5.8 (at least until my current work finished).

**Comment by RadhiFadlillah** [comment]
Closed in favor of #508.
